ams/metric-reporter/prometheus-reporter/src/main/java/com.netease.arctic.metrics.reporter.promethues/MetricCollector.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netease.arctic.metrics.reporter.promethues;
+
+import com.netease.arctic.ams.api.metrics.Counter;
+import com.netease.arctic.ams.api.metrics.Gauge;
+import com.netease.arctic.ams.api.metrics.Metric;
+import com.netease.arctic.ams.api.metrics.MetricDefine;
+import com.netease.arctic.ams.api.metrics.MetricKey;
+import com.netease.arctic.ams.api.metrics.MetricSet;
+import com.netease.arctic.ams.api.metrics.MetricType;
+import io.prometheus.client.Collector;
+import org.apache.iceberg.relocated.com.google.common.collect.Lists;
+
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+/** Metric type converter for prometheus api */
+public class MetricCollector extends Collector {
+  private static final String PREFIX = "amoro_";
+  private static final Pattern NAME_PATTERN = Pattern.compile("[a-zA-Z_:][a-zA-Z0-9_:]*");
+  private static final Pattern LABEL_PATTERN = Pattern.compile("[a-zA-Z_][a-zA-Z0-9_]*");
+  MetricSet metrics;
+
+  public MetricCollector(MetricSet metrics) {
+    this.metrics = metrics;
+  }
+
+  @Override
+  public List<MetricFamilySamples> collect() {
+    Map<MetricKey, Metric> registeredMetrics = metrics.getMetrics();
+
+    Map<MetricDefine, List<MetricKey>> metricDefineMap =
+        registeredMetrics.keySet().stream()
+            .collect(
+                Collectors.groupingBy(
+                    MetricKey::getDefine,
+                    Collectors.mapping(Function.identity(), Collectors.toList())));
+    return metricDefineMap.entrySet().stream()
+        .filter(entry -> isValidMetric(entry.getKey()))
```

Comment:
Should we logging the filter out metrics, it's more friendly to troubleshooting scenarios with missing metrics

Reply:
fixed.